### PR TITLE
Enh([Dshboard]):Status grid display issue when reducing screen size

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
**Description**:  fix issue when reducing screen size

**video** 

[screen-capture (7).webm](https://github.com/user-attachments/assets/957940f8-9e3d-45dc-b20b-090b2bbb66f7)

### CODE REVIEW

File: centreon/packages/ui/src/Graph/HeatMap/model.ts [hunk 1]
```diff
@@ -19,7 +21,9 @@ export interface HeatMapProps<TData> {
     backgroundColor,
     id,
     data,
-    isSmallestSize
+    isSmallestSize,
+    tileSize,
+    isMediumSize
```
[the remark this answers is not in the document]
Comment (Noha-ElAbrouki) [reply centreon/packages/ui/src/Graph/HeatMap/model.ts:26]: those parameters are optional , bedise tooltip donit need this props

**Review by quborg — APPROVED** [review]

File: centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx [hunk 2]
```diff
@@ -60,6 +69,11 @@ const ResponsiveHeatMap = <TData,>({
   }, [width, tiles, height]);
 
   const isSmallestSize = equals(tileSize, smallestTileSize);
+  const isMediumSize = !isSmallestSize && lt(tileSize, 90);
+
+  useEffect(() => {
+    previouseTileSize.current = tileSize;
+  }, [tileSize]);
```
Comment (Thebarda) [inline centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx:76]:
```suggestion
```
you don't have to use useEffect that update a ref

File: centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { useMemo } from 'react';
+import { useEffect, useMemo, useRef } from 'react';
```
Comment (Thebarda) [inline centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx:1]:
```suggestion
import { useMemo, useRef } from 'react';
```

File: centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx [hunk 4]
```diff
@@ -44,6 +46,13 @@ const ResponsiveHeatMap = <TData,>({
     const theoricalTotalTilesWidth =
       tilesLength * tileWidth + (tilesLength - 1) * gap;
 
+    const canUpdateTileSize =
+      Math.abs(tileWidth - previouseTileSize.current) > toleratedRangeWidth;
+
+    if (!canUpdateTileSize) {
+      return previouseTileSize.current;
```
Comment (Thebarda) [inline centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx:53]:
```suggestion
      Math.abs(tileWidth - previousTileSize.current) > toleratedRangeWidth;

    if (!canUpdateTileSize) {
      return previousTileSize.current;
```

File: centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx [hunk 5]
```diff
@@ -28,6 +29,7 @@ const ResponsiveHeatMap = <TData,>({
   width: number;
 }): JSX.Element | null => {
   const { classes, cx } = useHeatMapStyles();
+  const previouseTileSize = useRef(0);
```
Comment (Thebarda) [inline centreon/packages/ui/src/Graph/HeatMap/ResponsiveHeatMap.tsx:32]:
```suggestion
  const previousTileSize = useRef(0);
```

**Review by Thebarda — DISMISSED** [review]

**Review by quborg — APPROVED** [review]

**Pull request merged**